core/unit_test/CMakeLists.txt

```diff
@@ -128,7 +128,8 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;OpenMPTarget;HIP;SYCL)
         HostSharedPtrAccessOnDevice
         QuadPrecisionMath
         ExecSpacePartitioning
-        MathematicalSpecialFunctions
+        MathematicalFunctions1
+        MathematicalFunctions2
```

Comment:
Certainly you meant to replace `MathematicalFunctions` not `MathematicalSpecialFunctions`.
You just signed up for duplicated symbols errors at link-time.